spatialdata/_core/_spatial_query.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+import numpy as np
+import pyarrow as pa
+
+from spatialdata._core.coordinate_system import CoordinateSystem
+
+if TYPE_CHECKING:
+    pass
+
+
+@dataclass(frozen=True)
+class BaseSpatialRequest:
+    """Base class for spatial queries."""
+
+    coordinate_system: CoordinateSystem
+
+    def __post_init__(self) -> None:
+        # validate the coordinate system
+        spatial_axes = _get_spatial_axes(self.coordinate_system)
+        if len(spatial_axes) == 0:
+            raise ValueError("No spatial axes in the requested coordinate system")
+
+
+@dataclass(frozen=True)
+class BoundingBoxRequest(BaseSpatialRequest):
+    """Query with an axis-aligned bounding box.
+
+    Attributes
+    ----------
+    coordinate_system : CoordinateSystem
+        The coordinate system the coordinates are expressed in.
+    min_coordinate : np.ndarray
+        The coordinate of the lower left hand corner (i.e., minimum values)
+        of the bounding box.
+    max_coordiate : np.ndarray
+        The coordinate of the upper right hand corner (i.e., maximum values)
+        of the bounding box
+    """
+
+    min_coordinate: np.ndarray  # type: ignore[type-arg]
+    max_coordinate: np.ndarray  # type: ignore[type-arg]
+
+
+def _get_spatial_axes(
```

Comment:
I'll move this as a method in `CoordinateSystem`

Reply:
Done!